core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -169,12 +168,8 @@ public InternalEngine(EngineConfig engineConfig) throws EngineException {
                 }
             }
         }
-        for (EngineConfig.EngineCreationListener listener : engineConfig.getEngineCreationListeners()) {
-            listener.engineCreated(this);
-        }
-        for (ReferenceManager.RefreshListener listener : engineConfig.getRefreshListeners()) {
-            searcherManager.addListener(listener);
-        }
+        searcherManager.addListener(engineConfig.getRefreshListeners());
+        engineConfig.getRefreshListeners().setTranslog(translog);
```

Comment:
also please move this into the finally block above so we release store locks etc if we fail.